### PR TITLE
Code Insights" Migrate live preview handler to hooks

### PR DESCRIPTION
**Pull request by vovakulikov**
Part of https://github.com/sourcegraph/sourcegraph/issues/31085

## Background 

In this PR we migrate live preview handlers (search-based, capture group, compute based insights live previews) to hooks, previously they were built as a part of code insights backend service. For more context on why we should remove backend service class read in [the tracking issue](https://github.com/sourcegraph/sourcegraph/issues/41145) but TL;DR live preview hook should be easy to reuse which is important if we would want to integrate insights with other UI. 

## Test plan
- Manually test search based, capture group and compute based insights creation UI. 
- Check that live preview card doesn't have any regressions in lights of data fetching states (loading, error, actually data, intact state when we have blur chart and initial message)
- Check dynamic insight preview card on the getting started insight page.

## App preview:

- [Web](https://sg-web-vk-migrate-live-preview-to-hooks.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.00% (+0.30 kb) | 0.00% (+0.30 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/vk/migrate-live-preview-to-hooks/compare-report.html;tab=live_object) for a full comparison between the commits 82101ee and 57ebf1a or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/web/src/enterprise/insights/core/hooks/live-preview-insight/index.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export type { Datum, SeriesWithStroke } from './use-live-preview-series-insight'
+export { useLivePreviewSeriesInsight, GET_INSIGHT_PREVIEW_GQL } from './use-live-preview-series-insight'
+
+export { useLivePreviewComputeInsight } from './use-live-preview-compute-insight'
```
Comment (chwarwick) [inline client/web/src/enterprise/insights/core/hooks/live-preview-insight/index.ts:4]: The naming on this has been a little confusing to me because capture groups is compute too.  Would you considering calling this `useLivePreviewGroupByInsight`?

File: client/wildcard/src/components/Charts/components/line-chart/LineChart.tsx [hunk 2]
```diff
@@ -69,6 +69,7 @@ export interface LineChartProps<Datum> extends SeriesLikeChart<Datum>, SVGProps<
 }
 
 const identity = <T,>(argument: T): T => argument
+const DEFAULT_LINE_CHART_PADDING = { top: 16, right: 18, bottom: 20, left: 0 }
```
Comment (chwarwick) [inline client/wildcard/src/components/Charts/components/line-chart/LineChart.tsx:72]: the changes to this file appear unrelated what is its purpose? Should it be separate?

**Review by chwarwick — COMMENTED** [review]
I think it's unlikely that we would reuse the live preview elsewhere in the app but in general I think this refactor cleans things up and makes it easier to work with if we need to in the future.

File: client/wildcard/src/components/Charts/components/line-chart/LineChart.tsx (hunk 2 above)
Comment (vovakulikov) [reply client/wildcard/src/components/Charts/components/line-chart/LineChart.tsx:72]: @chwarwick you're right about this, I just didn't want to open yet another PR for this small tweak (since this PR is about live preview I thought that this makes sense to include this small fix).

If you think that it adds a big risk to break something and reverting it later will be difficult compared to when these changes were in a different PR I can pull them out from this PR. Otherwise I would still keep it here.

File: client/web/src/enterprise/insights/core/hooks/live-preview-insight/index.ts (hunk 1 above)
Comment (vovakulikov) [reply client/web/src/enterprise/insights/core/hooks/live-preview-insight/index.ts:4]: @chwarwick It requires a bigger change. We have `compute` all over the plain on the frontend. In fact our FE codebase naming doesn't match with product naming (because we changed insight types names at some point) 

I would say that having `useLivePreviewGroupByInsight` and having `<Compute ...>` files/entities in other places is worse than having wrong but consistent naming in all places. 

I will open an issue for aligning types and product naming.

**Comment by vovakulikov** [comment]
@chwarwick I agree and disagree about live preview outside of code insights, I've found live preview insights very powerful instrument in prototyping, for example we probably could add insights-like information on the repository page, like language stats insight (similar to what GH has on their repo pages) or compute insights with information about most popular files in the repo.

File: client/wildcard/src/components/Charts/components/line-chart/LineChart.tsx (hunk 2 above)
Comment (chwarwick) [reply client/wildcard/src/components/Charts/components/line-chart/LineChart.tsx:72]: I don't think it's a big risk considering that charts aren't used in many places, but its best to keep things like this separate. These reviews are usually large and unrelated changes add to the complexity of whats changing.  Also in cases were it's changes to a shared library like wildcard components I think you should be extra careful to keep those separate so that there is a cleaner version history of those components incase they need to be reverted.

File: client/web/src/enterprise/insights/core/hooks/live-preview-insight/index.ts (hunk 1 above)
Comment (chwarwick) [reply client/web/src/enterprise/insights/core/hooks/live-preview-insight/index.ts:4]: Thats fine, and it's probably more overhead for me because I know what powers those insights and when I think compute I think capture group first and then group by a distant second.
